app/services/proxy_status.rb

```diff
@@ -26,10 +26,24 @@ def tftp_server
     end
   end
 
+  # TODO: extract to another status implementation
+  def environment_stats
+    raise ::Foreman::Exception.new(N_('No Puppet feature for %s'), proxy.to_label) unless proxy.has_feature?('Puppet')
+    Rails.cache.fetch("proxy_#{proxy.id}/environments", :expires_in => cache_duration) do
+      fetch_proxy_data do
+        proxy = ProxyAPI::Puppet.new(:url => @proxy.url)
+        env_counts = {}
+        proxy.environments.each{ |env| env_counts[env] = proxy.class_count(env) }
+        env_counts
+      end
+    end
+  end
+
   def revoke_cache!
     # As memcached does not support delete_matched, we need to delete each
     Rails.cache.delete(versions_cache_key)
     Rails.cache.delete("proxy_#{proxy.id}/tftp_server")
+    Rails.cache.delete("proxy_#{proxy.id}/environments")
```

Comment:
`puppet_environments` rather, just so we can have more config management system environments